### PR TITLE
Remove extra difficulty check for proofs

### PR DESCRIPTION
**Pull request by tjsturos**
There was a redundant check for if the increment was over 800k. 

This would be redundant due to the first check of "if the difficulty was under 25k" would be met at anything over 700k.

Moved to a function, due to it being used twice (in the Calculating and Verifying of the Proof) and to be able to add tests.

### CODE REVIEW

**Review by CassOnMars — APPROVED** [review]

File: node/crypto/wesolowski_frame_prover.go [hunk 1]
```diff
@@ -579,15 +579,24 @@ func (w *WesolowskiFrameProver) VerifyWeakRecursiveProof(
 	}
 }
 
+func (w *WesolowskiFrameProver) CalculateChallengeProofDifficulty(
+	increment uint32,
+) uint32 {
+	difficulty := 200000 - (increment / 4)
```
Comment (CassOnMars) [inline node/crypto/wesolowski_frame_prover.go:585]: the increment check is needed because this is a uint32, it'll underflow

**Comment by CassOnMars** [comment]
looks good minus the underflow check

File: node/crypto/wesolowski_frame_prover.go (hunk 1 above)
Comment (tjsturos) [reply node/crypto/wesolowski_frame_prover.go]: Reordered the function to only do the calculation on increments less than 700k

**Review by CassOnMars — APPROVED** [review]
awesome, thanks!

**Pull request merged**